mmengine/config/config.py

```diff
@@ -288,6 +290,29 @@ def _substitute_predefined_vars(filename: str, temp_config_name: str):
         with open(temp_config_name, 'w', encoding='utf-8') as tmp_config_file:
             tmp_config_file.write(config_file)
 
+    @staticmethod
+    def _substitute_environment_vars(filename: str, temp_config_name: str):
+        """Substitute environment variables in config with actual values.
+
+        Args:
+            filename (str): Filename of config.
+            temp_config_name (str): Temporary filename to save substituted
+                config.
+        """
+        with open(filename, encoding='utf-8') as f:
+            config_file = f.read()
+        regexp = r'\{\{[\'\"]?\s*\$(\w+)\s*\:?\s*(\S*?)\s*[\'\"]?\}\}'
```

Comment:
is [\'\"] necessary?

Reply:
This part is to handle the case when the environment variable is not used in str, like the **Example 2** above. 

`{{$NUM_CLASSES:80}}` is also work well. But the config will report a syntax error if `$NUM_CLASSES:80` is not wrapped by `''`.

Reply:
It seems `?` after `:` is not necessary, otherwise {{$ITEM11 123}} will also be matched.